src/features/session_replay/aggregate/index.js

```diff
@@ -211,24 +213,25 @@ export class Aggregate extends AggregateBase {
     const info = getInfo(this.agentIdentifier)
     return {
       qs: {
-        protocol_version: '0',
-        content_encoding: 'gzip',
-        browser_monitoring_key: info.licenseKey
-      },
-      body: {
+        browser_monitoring_key: info.licenseKey,
         type: 'SessionReplay',
-        appId: Number(info.applicationID),
-        timestamp: Date.now(),
-        blob: JSON.stringify(this.events), // this needs to be a stringified JSON array of rrweb nodes
-        attributes: {
+        app_id: info.applicationID,
+        protocol_version: '0',
+        attributes: encodeObj({
+          ...(this.shouldCompress && { content_encoding: 'gzip' }),
+          'replay.firstTimestamp': this.timestamp.first,
+          'replay.lastTimestamp': this.timestamp.last,
+          'replay.durationMs': this.timestamp.last - this.timestamp.first,
+          agentVersion: agentRuntime.version,
           session: agentRuntime.session.state.value,
           hasSnapshot: this.hasSnapshot,
           hasError: this.hasError,
-          agentVersion: agentRuntime.version,
           isFirstChunk: this.isFirstChunk,
+          decompressedBytes: this.payloadBytesEstimation,
           'nr.rrweb.version': RRWEB_VERSION
-        }
-      }
+        }, MAX_PAYLOAD_SIZE - this.payloadBytesEstimation).substring(1) // remove the leading '&'
```

Comment:
not sure on the 30k part.  We used to have a limit of 2k for ie6, so its probably something outdated, or tied to BAM's query parsing rules.

Reply:
Right, I'm not talking about custom attributes, I mean as the number of attributes or spec expands in the future.

Reply:
Did you also take a look at the baseParams thing?

Reply:
if spec changes in the future we will have to consider sizing for sure.  shouldnt be an issue right now tho.

The base params have no effect here.  The attributes property itself is a url-encoded string, so it would have a leading `&` by using that method,

So in essence it will look like

```javascript
        browser_monitoring_key: <string>,
        type: <string>,
        app_id: <string>,
        protocol_version: <string>
        attributes: <url encoded string> <-- We dont want this url-encoded string to start with `&`
```

```javascript
?browser_monitoring_key=abcdefg&type=SessionReplay&app_id=hijklmnop&protocol_version=0&attributes=replay.firstTimestamp=1234&replay.lastTimestamp=5678&...etc
```